openedx_lti_tool_plugin/middleware.py

```diff
@@ -0,0 +1,96 @@
+"""Middleware for openedx_lti_tool_plugin."""
+import re
+from typing import Any, Callable
+
+from django.conf import settings
+from django.contrib.auth import logout
+from django.core.exceptions import MiddlewareNotUsed
+from django.http import HttpResponse
+from django.http.request import HttpRequest
+
+from openedx_lti_tool_plugin.models import LtiProfile
+
+
+class LtiViewPermissionMiddleware:
+    """LTI view permission middleware.
+
+    Attributes:
+        URL_WHITELIST (list): List of whitelisted URL patterns.
+        get_response (Any): Callable returned by previous middleware.
+    """
+
+    def __init__(self, get_response: Any):
+        """Middleware initialization.
+
+        Args:
+            get_response: Callable returned by previous middleware.
+
+        Raises:
+            MiddlewareNotUsed if plugin is disabled.
+        """
+        # Disable middleware is plugin is disabled.
+        if not getattr(settings, 'OLTITP_ENABLE_LTI_TOOL', False):
+            raise MiddlewareNotUsed()
+
+        self.get_response = get_response
+
+    def __call__(self, request: HttpRequest) -> HttpResponse:
+        """Process request.
+
+        Args:
+            request: HTTP request object.
+
+        Returns:
+            HTTP response object.
+        """
+        return self.get_response(request)
+
+    def process_view(
+        self,
+        request: HttpRequest,
+        view_func: Callable[..., str],
+        *args: tuple,
+    ):
+        """Process request before view call.
+
+        Args:
+            request: HTTP request object.
+            view_func: View function.
+            *args: Variable length argument list.
+        """
+        # Avoid processing the view if no LtiProfile is found.
+        if not LtiProfile.objects.filter(user=request.user.id).exists():
+            return None
+
+        # Block all views that match in the OLTITP_URL_BLOCKLIST patterns setting.
+        if any(re.match(regex, request.path) for regex in settings.OLTITP_URL_BLOCKLIST):
+            return logout(request)
+
+        # List of URL patterns to allow.
+        url_whitelist = [
```

Comment:
What do you think of defining these default urls patterns in a constant? a then join the constant list and the settings list here